storage/innobase/handler/ha_innodb.cc

```diff
@@ -3860,6 +3860,16 @@ static int innodb_init_params()
 		}
 	}
 
+	if (srv_undo_tablespaces + 3 > innobase_open_files) {
+		sql_print_warning(
+			"InnoDB: innodb_open_files %ld should be "
+			"greater than total number of undo tablespace, "
+			"system & temporary tablespace. So changing "
+			"the value of innodb_open_files to %ld",
+			innobase_open_files, srv_undo_tablespaces + 3);
```

Comment:
I’d suggest to use some more `=` in the message so that it becomes easier to understand for non-native English speakers:
```
InnoDB: innodb_open_files=%ld is smaller than 3+innodb_undo_tablespaces=%ld; adjusting to innodb_open_files=%ld
```
The message does not necessarily need to explain what the 3 is coming from.

By the way, should this take into account system or temporary tablespaces that consist of multiple files? Can you cover that this in the test case?